src/Execution/Arguments/SaveModel.php

```diff
@@ -66,9 +66,15 @@ public function __invoke($model, $args)
         $model->save();
 
         if ($this->parentRelation instanceof BelongsTo) {
-            $this->parentRelation
-                ->associate($model)
-                ->save();
+            $parentModel = $this->parentRelation
+                                ->associate($model);
+
+            // If the parent Model does not exist (still to be saved),
+            // a save could break any pending belongsTo relations that still
+            // needs to be created and associated with the parent model
+            if ($parentModel->exists()) {
+                $parentModel->save();
```

Comment:
That would trigger multiple saves, right? Is there a simple way through which we can save just once?

Reply:
Hi, this actually reduces the number of saves being done before this change. Before it will always save the parentModel. I tried to remove the save method, but that breaks a test case for deep nested mutations when using a sync. The test that broke was testUpsertAcrossTwoNestedBelongsToRelationsAndOverrideExistingModel on line 907. I found that test a little strange, since it changes the id of the related user to the role, using a hasMany, when you are triggering that change in a mutation to the user.

Reply:
I looked into that testcase, and what it does is run a deep nested sync after a many to many relation being used to create a role. If i remove the save, in that case the sync would be overwritten by the first many to many relation:

User -> hasMany -> role_user -> belongsTo -> role -> belongsToMany -> users
S1                                  S4                                     S2                                       S3

Where Sn is the order of saves that the actual code does.

I am not sure the usefulness of this case, where you want to create a model related to an existing one, and then change that relation. Could be simpler to just create the Role and then associated the Users?

Reply:
The save is only needed to guarantee that a operation like that persists before applying changes to the same relation in deeper levels of the mutation. Once again, i am not sure the usefulness of this specific case. If such redeems unnecessary, the save could be removed. The problem is that nothing tell us that this specific case is not used by other users.

Reply:
That test case is indeed a bit far fetched, but i think it was added by someone who had that exact use case. The other test that fails when removing `save()` here is one where `@create` is used as a nested arg resolver.

The check you implemented makes the most use cases possible and work smoothly, so i am fine with adding it.